design/open/spark-outputcommitter/lakeFS-output-committer-execution-plan.md

```diff
@@ -0,0 +1,38 @@
+# LakeFS Output Committer - Execution Plan
+[LakeFS Output Committer Proposal](https://github.com/treeverse/lakeFS/blob/master/design/open/spark-outputcommitter/committer.md)
+
+## Overview
+Implementing lakeFS Output Committer meaning implementing the class [FileOutputCommitter](https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/mapreduce/lib/output/FileOutputCommitter.html) and using lakeFS client to create/merge branches as part of setup and commit job/tasks (this is the [sample flow](https://github.com/treeverse/lakeFS/blob/master/design/open/spark-outputcommitter/committer.md#sample-flow))
+
+### Milestone 1
+The first milestone is focused on writing a file with single mode (probably text) and one file format (probably append).
+At this stage we'll implement `commit`, `setup`, `abort` class methods.
+
+### Milestone 2
+This milstone is focused on writing all file formats, in all modes. Also, implement the relevant remaining class methods.
```

Comment:
> Also, implement the relevant remaining class methods.

Do we already know what are the relevant class method? or those that will be needed to support all formats in all modes considered relevant?

Reply:
I think this should be determined when we get there.